File: src/main/java/ru/octol1ttle/flightassistant/alerts/impl/autoflight/AutoThrustOffAlert.java
package ru.octol1ttle.flightassistant.alerts.impl.autoflight;

import net.minecraft.client.font.TextRenderer;
import net.minecraft.client.gui.DrawContext;
import net.minecraft.text.Text;
import org.jetbrains.annotations.NotNull;
import ru.octol1ttle.flightassistant.DrawHelper;
import ru.octol1ttle.flightassistant.alerts.api.BaseAlert;
import ru.octol1ttle.flightassistant.alerts.api.IECAMAlert;
import ru.octol1ttle.flightassistant.alerts.impl.AlertSoundData;
import ru.octol1ttle.flightassistant.computers.impl.autoflight.AutoFlightController;
import ru.octol1ttle.flightassistant.config.FAConfig;
import ru.octol1ttle.flightassistant.registries.ComputerRegistry;

public class AutoThrustOffAlert extends BaseAlert implements IECAMAlert {
    private final AutoFlightController autoflight = ComputerRegistry.resolve(AutoFlightController.class);

    public AutoThrustOffAlert() {
        this.hidden = true;
    }

    @Override
    public boolean isTriggered() {
        return !autoflight.autoThrustEnabled;
    }

    @Override
    public @NotNull AlertSoundData getSoundData() {
        return autoflight.athrDisconnectionForced ? AlertSoundData.MASTER_CAUTION : AlertSoundData.EMPTY;
    }

    @Override
    public int render(TextRenderer textRenderer, DrawContext context, int x, int y, boolean highlight) {
        return DrawHelper.drawHighlightedText(textRenderer, context, Text.translatable("alerts.flightassistant.autoflight.auto_thrust_off"), x, y,
                FAConfig.indicator().cautionColor,
                highlight && autoflight.athrDisconnectionForced
        );
    }
}

File: src/main/kotlin/ru/octol1ttle/flightassistant/impl/alert/autoflight/AutoThrustOffAlert.kt
package ru.octol1ttle.flightassistant.impl.alert.autoflight

import net.minecraft.client.gui.DrawContext
import net.minecraft.text.Text
import ru.octol1ttle.flightassistant.api.alert.Alert
import ru.octol1ttle.flightassistant.api.alert.AlertData
import ru.octol1ttle.flightassistant.api.alert.ECAMAlert
import ru.octol1ttle.flightassistant.api.computer.ComputerView
import ru.octol1ttle.flightassistant.api.util.extensions.cautionColor
import ru.octol1ttle.flightassistant.api.util.extensions.drawText

class AutoThrustOffAlert(computers: ComputerView) : Alert(computers), ECAMAlert {
    override val data: AlertData = AlertData.MASTER_CAUTION
    override val priorityOffset: Int = 20

    override fun shouldActivate(): Boolean {
        return computers.automations.autoThrustAlert
    }

    override fun render(drawContext: DrawContext, firstLineX: Int, otherLinesX: Int, firstLineY: Int): Int {
        return drawContext.drawText(Text.translatable("alerts.flightassistant.autoflight.auto_thrust_off"), firstLineX, firstLineY, cautionColor)
    }
}
